// @chatui/core/lib/components/MessageContainer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageContainer = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _PullToRefresh = require("../PullToRefresh");

var _Message = require("../Message");

var _canUse = _interopRequireDefault(require("../../utils/canUse"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/* eslint-disable no-underscore-dangle */
var listenerOpts = (0, _canUse.default)('passiveListener') ? {
  passive: true
} : false;

var MessageContainer = /*#__PURE__*/_react.default.forwardRef(function (props, ref) {
  var messages = props.messages,
      loadMoreText = props.loadMoreText,
      onRefresh = props.onRefresh,
      onScroll = props.onScroll,
      renderBeforeMessageList = props.renderBeforeMessageList,
      renderMessageContent = props.renderMessageContent;
  var messagesRef = (0, _react.useRef)(null);
  var scroller = (0, _react.useRef)(null);
  var lastMessage = messages[messages.length - 1];
  (0, _react.useEffect)(function () {
    var wrapper = scroller.current.wrapperRef.current;
    var animated = !!wrapper.scrollTop;
    scroller.current.scrollToEnd({
      animated: animated
    });
  }, [lastMessage]);
  (0, _react.useEffect)(function () {
    var wrapper = messagesRef.current;
    if (!wrapper) return undefined;
    var needBlur = false;
    var startY = 0;

    function reset() {
      needBlur = false;
      startY = 0;
    }

    function touchStart(e) {
      var _document = document,
          activeElement = _document.activeElement;

      if (activeElement && activeElement.nodeName === 'TEXTAREA') {
        needBlur = true;
        startY = e.touches[0].clientY;
      }
    }

    function touchMove(e) {
      if (needBlur && Math.abs(e.touches[0].clientY - startY) > 20) {
        document.activeElement.blur();
        reset();
      }
    }

    wrapper.addEventListener('touchstart', touchStart, listenerOpts);
    wrapper.addEventListener('touchmove', touchMove, listenerOpts);
    wrapper.addEventListener('touchend', reset);
    wrapper.addEventListener('touchcancel', reset);
    return function () {
      wrapper.removeEventListener('touchstart', touchStart);
      wrapper.removeEventListener('touchmove', touchMove);
      wrapper.removeEventListener('touchend', reset);
      wrapper.removeEventListener('touchcancel', reset);
    };
  }, []);
  (0, _react.useImperativeHandle)(ref, function () {
    return {
      ref: messagesRef,
      scrollToEnd: scroller.current.scrollToEnd
    };
  }, []);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "MessageContainer",
    ref: messagesRef,
    tabIndex: -1
  }, renderBeforeMessageList && renderBeforeMessageList(), /*#__PURE__*/_react.default.createElement(_PullToRefresh.PullToRefresh, {
    onRefresh: onRefresh,
    onScroll: onScroll,
    loadMoreText: loadMoreText,
    ref: scroller
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "MessageList"
  }, messages.map(function (msg) {
    return /*#__PURE__*/_react.default.createElement(_Message.Message, (0, _extends2.default)({}, msg, {
      renderMessageContent: renderMessageContent,
      key: msg._id
    }));
  }))));
});

exports.MessageContainer = MessageContainer;
